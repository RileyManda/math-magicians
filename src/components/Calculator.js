import React, { useState } from 'react';
import calculate from '../logic/calculate';
import operate from '../logic/operate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setState((prevState) => {
      const newState = calculate(prevState, buttonName, operate);
      return newState;
    });
  };

  return (
    <div className="Container">
      <div className="calcInput">
        <input type="text" value={state.next || state.total || '0'} />
      </div>
      <div className="BodyContainer">
        <div className="calcBodyleft">
          <div className="LeftButtonContainer">
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('AC')}
            >
              AC
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('+/-')}
            >
              +/-
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('%')}
            >
              %
            </button>
            {' '}
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('7')}
            >
              7
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('8')}
            >
              8
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('9')}
            >
              9
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('4')}
            >
              4
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('5')}
            >
              5
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('6')}
            >
              6
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('1')}
            >
              1
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('2')}
            >
              2
            </button>
            <button
              type="button"
              className="leftButtons"
              onClick={() => handleClick('3')}
            >
              3
            </button>
            <div className="FooterButtons">
              <button
                type="button"
                className="zero FooterButtons"
                onClick={() => handleClick('0')}
              >
                0
              </button>
              <button
                type="button"
                className="fullstop FooterButtons"
                onClick={() => handleClick('.')}
              >
                .
              </button>
            </div>
          </div>
        </div>
        <div className="calcBodyright">
          <div className="RightButtonContainer">
            <button
              type="button"
              className="rightButtons"
              onClick={() => handleClick('รท')}
            >
              รท
            </button>
            <button
              type="button"
              className="rightButtons"
              onClick={() => handleClick('x')}
            >
              ร
            </button>
            <button
              type="button"
              className="rightButtons"
              onClick={() => handleClick('-')}
            >
              -
            </button>
            <button
              type="button"
              className="rightButtons"
              onClick={() => handleClick('+')}
            >
              +
            </button>
            <button
              type="button"
              className="rightButtons"
              onClick={() => handleClick('=')}
            >
              =
            </button>
            {' '}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
