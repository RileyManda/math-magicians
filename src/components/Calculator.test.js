import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Calculator from './Calculator';

jest.mock('../logic/calculate');

describe('Calculator', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });

  it('should update state on input change', () => {
    render(<Calculator />);
    const input = screen.getByRole('textbox');

    fireEvent.change(input, { target: { value: '0' } });

    expect(parseFloat(input.value)).toBe(0);
  });
});
