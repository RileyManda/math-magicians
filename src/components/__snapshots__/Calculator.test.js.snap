// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match the snapshot 1`] = `
<div>
  <div
    class="CalculatorSection"
  >
    <div
      class="CalcHeading"
    >
      <h2>
        Lets do some math!
      </h2>
    </div>
    <div
      class="CalcContainer"
    >
      <div
        class="calcInput"
      >
        <input
          id="input"
          type="text"
          value="0"
        />
      </div>
      <div
        class="CalcBodyContainer"
      >
        <div
          class="calcBodyleft"
        >
          <div
            class="LeftButtonContainer"
          >
            <button
              class="leftButtons"
              type="button"
            >
              AC
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              +/-
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              %
            </button>
             
            <button
              class="leftButtons"
              type="button"
            >
              7
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              8
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              9
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              4
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              5
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              6
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              1
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              2
            </button>
            <button
              class="leftButtons"
              type="button"
            >
              3
            </button>
            <div
              class="FooterButtons"
            >
              <button
                class="zero FooterButtons"
                type="button"
              >
                0
              </button>
              <button
                class="fullstop FooterButtons"
                type="button"
              >
                .
              </button>
            </div>
          </div>
        </div>
        <div
          class="calcBodyright"
        >
          <div
            class="RightButtonContainer"
          >
            <button
              class="rightButtons"
              type="button"
            >
              รท
            </button>
            <button
              class="rightButtons"
              type="button"
            >
              ร
            </button>
            <button
              class="rightButtons"
              type="button"
            >
              -
            </button>
            <button
              class="rightButtons"
              type="button"
            >
              +
            </button>
            <button
              class="rightButtons"
              type="button"
            >
              =
            </button>
             
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
